src/Symfony/Bundle/FrameworkBundle/Command/TranslationUpdateCommand.php

```diff
@@ -203,8 +211,7 @@ protected function execute(InputInterface $input, OutputInterface $output)
             if (!$bundleTransPath) {
                 $bundleTransPath = end($transPaths).'translations';
             }
-
-            $writer->writeTranslations($operation->getResult(), $input->getOption('output-format'), array('path' => $bundleTransPath, 'default_locale' => $this->getContainer()->getParameter('kernel.default_locale')));
+            $writer->writeTranslations($operation->getResult($input->getOption('domain')), $input->getOption('output-format'), array('path' => $bundleTransPath, 'default_locale' => $this->getContainer()->getParameter('kernel.default_locale')));
```

Comment:
@nicolas-grekas I added an optional one in src/Symfony/Component/Translation/Catalogue/AbstractOperation.php